shell/browser/api/electron_api_web_contents_view.cc

```diff
@@ -78,38 +57,78 @@ WebContentsView::~WebContentsView() {
   }
 }
 
+gin::Handle<WebContents> WebContentsView::GetWebContents(v8::Isolate* isolate) {
+  return gin::CreateHandle(isolate, api_web_contents_);
+}
+
 void WebContentsView::WebContentsDestroyed() {
   api_web_contents_ = nullptr;
   web_contents_.Reset();
 }
 
+// static
+gin::Handle<WebContentsView> WebContentsView::Create(
+    v8::Isolate* isolate,
+    const gin_helper::Dictionary& web_preferences) {
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  v8::Local<v8::Value> arg = gin::ConvertToV8(isolate, web_preferences);
+  v8::Local<v8::Object> obj;
+  if (GetConstructor(isolate)->NewInstance(context, 1, &arg).ToLocal(&obj)) {
```

Comment:
Why does `Create` have to go through the V8 binding to call `New` and wrap the handle ? If the call is only happening on c++ land, can it not directly call `New` and wrap the result ?

Reply:
Calling `New` is equivalent to calling `WebContentsView()` in JS which does not allocate a holder, we have to call `new WebContentsView()` in JS and the only way to do that in JS is to call the `v8::Function::NewInstance` API.

Reply:
Classes based on `gin::Wrappable` does not need to do this because the objects are generated from `v8::ObjectTemplate`, while the `View` classes have to use `v8::FunctionTemplate` to implement prototype inheritance.

Reply:
We should be able to call New directly once we figured out a way to move the `View` classes to `gin::Wrappable`, but currently a `Create` method is required.